src/main/java/org/jabref/logic/importer/fetcher/IssnFetcher.java

```diff
@@ -0,0 +1,80 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.IdBasedFetcher;
+import org.jabref.logic.importer.IdFetcher;
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.integrity.ISSNChecker;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.identifier.ISSN;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Fetcher to generate the BibTex entry from an ISSN.
+ * The idea is to use the {@link DOAJFetcher} to do a request for a given ISSN number.
+ */
+
+public class IssnFetcher implements IdBasedFetcher, IdFetcher<ISSN> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IssnFetcher.class);
+    private final DOAJFetcher doajFetcher;
+    private final String SEARCH_URL = "https://doaj.org/api/search/journals/";
+    private final ISSNChecker issnChecker;
+    private final ImportFormatPreferences importFormatPreferences;
+
+    public IssnFetcher(ImportFormatPreferences importFormatPreferences) {
+        this.importFormatPreferences = importFormatPreferences;
+        this.doajFetcher = new DOAJFetcher(importFormatPreferences);
+        this.issnChecker = new ISSNChecker();
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+
+        Optional<String> checkedId = issnChecker.checkValue(identifier);
+        if (checkedId.isEmpty()) {
+            LOGGER.warn("Not a valid ISSN");
+            return Optional.empty();
+        }
+
+        Optional<BibEntry> bibEntry = Optional.empty();
+
+        String queryString = concatenateIssnWithId(identifier);
+        List<BibEntry> bibEntries = doajFetcher.performSearch(queryString);
+
+        if (!bibEntries.isEmpty()) {
```

Comment:
You could always return the first found 
e.e. bibEntries.stream().findfFirst() that already gives you an optional that you can simply return

Reply:
Ok, done. How do you think we can implement this Fecther to manually test in a running Jabref? Should we follow the original solution and implement a button near the ISSN field in Optional Fields?

Reply:
Before trying to test in a running JabRef, I will write the Unit Test for the Class.

Reply:
I would make it similar to the ISBN case

Reply:
Yep. And it should already be added automatically if you just add the fetcher to the list: 
https://github.com/JabRef/jabref/blob/9739619128738f5f4f99e3a0973fc1c5bb6de434/src/main/java/org/jabref/logic/importer/WebFetchers.java#L126-L149 

And add it to the IdentifierEditor class, then it will be avaiable here: 
<img width="523" alt="grafik" src="https://github.com/JabRef/jabref/assets/320228/f7f8e8ad-f5b4-4de4-a21e-c60a55d10046">

Reply:
ISSN is on the Select entry type, but it's not on the "update with bibliographic information from the web" that you showed and I think it's because the Issn Fetcher does not extend the EntryBasedFetcher class, should we implement this?

And as I expected it was not possible to search. When I executed the unit tests, it fails, always returning an Optional.empty(). I debugged the code to see what was happening and saw that for the ISSN like "2579-5341", it recognizes it as a not valid checksum. When I took off the hyphen ("25795341") it recognizes an invalid format. I only tested with 2 ISSN numbers, I'll test with some more.

Reply:
For the second one, Update with bib info from the web: Needs to be integrated here:

https://github.com/JabRef/jabref/blob/9739619128738f5f4f99e3a0973fc1c5bb6de434/src/main/java/org/jabref/gui/fieldeditors/identifier/IdentifierEditor.java#L44-L46


Regarding the ISSN checksum: 
Probably the issn class needs to account for hyphens as well Feel free to adjus that (tests welcome)

Reply:
Sure, I'll integrate it into IdentifierEditor and take a look at this ISSN checksum as well.

Reply:
Yesterday I made a mistake about the checksum of  ISSN, it was all correct. My mistake was not considering the return of a valid ISSN number as an Optional.empty(), so the isPresent method on the if condition was useless. I got it adjusted today, and now it's working. The next problem that I'm working on is, the DOAJFetcher is returning a NullPointerException because the bibEntryPreferences() is null.

Reply:
In the test you probably need to add another mock:

```

ImportFormatPreferences importPrefs = mock(ImportFormatPreferences.class)
BibEntryPreferences bibEntryPrefs = mock(BibEntryPrefernces.class)
when(importPrefs.getBIbEntryPreferences().thenReturn(bibEntryPrefs)
```

Reply:
And it worked like a charm! But now we have a "problem" that can be a future possibility. If we add an entry based on the ISSN number, it will return to us an Article. That is wrong but expected because we are using the DOAJFetcher which uses a URL that searches for articles (https://doaj.org/api/v1/search/articles/). I tested the outputs and we have multiple articles under the same ISSN. What we have to do is use another Search URL (https://doaj.org/api/search/journals/) to return Journal info, and later, we can also create another method that can return a List of the entries that are under the same ISSN number.

Reply:
Great to hear. Okay, I was not aware of this, but this makes sense, the ISSN is unique per journal issue/number but not per article. 
Ah okay, for completing the entry than we would need to use the journal endpoint to get at least the journal related metadata.

Yeah, I think the idea of getting all articles in a journal is also nice. However, then it cannot be used as an IDBsed Fetcher. Rather, it could work using the normal Search based fetcher

Reply:
About the journal endpoint, what should we do? Because the DOAJ Fetcher class is designed to deal with articles. Should we create a DOAJ Journal Fetcher?
And about getting all articles in a journal, I agree that using the Search based fetcher seems a way out.

Reply:
For a journal endpoint, there was just this PR https://github.com/JabRef/jabref/pull/10015  by @aqurilla 
Maybe he has an idea

Reply:
Just had a look at the issue https://github.com/JabRef/jabref/issues/10124, and I think most of the journal info fetching functionality using the ISSN is already implemented in `org.jabref.logic.journals.JournalInformationFetcher`. There is a button beside the ISSN field that shows this information as a popover. 
As per my understanding, the additional requirement for this issue is to take the journal info and merge it with the entry, similar to what is being done for ISBN -

<img width="2484" alt="image" src="https://github.com/JabRef/jabref/assets/12078268/d2134aef-a8ea-4c5b-aee4-c3d9095d83a8">

Reply:
There was a discussion about various API sources for fetching journal info, including DOAJ (https://github.com/JabRef/jabref/issues/6189#issuecomment-793617166), ultimately we decided to use a database hosted on JabRef's server which mainly contains data from Scimagojr (https://github.com/JabRef/jabref/pull/10015#issuecomment-1598253814)

Reply:
That's very interesting and a game-changer. I didn't get a detailed look at the journals package but I've already seen that a lot of things are already implemented. Some of the things we did here (almost everything) will not going to be used to implement the #10124 feature, but can be used to implement other features for the Articles session, right? I will take a detailed look at the journal package and the GUI parts of merging to see what to do next and start working on it.